### PR TITLE
feat: display new completed dialog

### PR DESCRIPTION
**Pull request by sjaanus**
1. Now the dialog will not close when SDK got connected
2. It will start to show the suggested production code. ( this will be attached in next PR)
3. Also, it has connected indicator on the right
4. Back button is removed in this stage.

image

### CODE REVIEW

File: frontend/src/component/onboarding/ConnectSdkDialog.tsx [hunk 1]
```diff
@@ -152,22 +156,25 @@ export const ConnectSdkDialog = ({
...
-                                <Button
-                                    variant='text'
-                                    color='inherit'
-                                    onClick={() => {
-                                        setStage('generate-api-key');
-                                    }}
-                                >
-                                    Back
-                                </Button>
+                                {!onboarded ? (
```
Comment (sjaanus) [inline frontend/src/component/onboarding/ConnectSdkDialog.tsx:159]: Hide the back button when onboarded.

File: frontend/src/component/onboarding/ConnectionInformation.tsx [hunk 2]
```diff
@@ -86,28 +86,38 @@ export const ConnectionInformation = ({
...
-                                height: 80,
-                            }}
-                            active={true}
-                        >
-                            <Pending fontSize='large' />
-                        </WhitePulsingAvatar>
-                    }
-                />
-            </ConnectionStatus>
+            {onboarded ? (
```
Comment (sjaanus) [inline frontend/src/component/onboarding/ConnectionInformation.tsx:89]: When onboarded, we have a new checkmark and text .

File: frontend/src/component/onboarding/SdkConnected.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    return <code>{children}</code>;
+};
+
+interface ISdkConnectedProps {
+    sdk: Sdk;
+}
+
+export const SdkConnected: FC<ISdkConnectedProps> = ({ sdk }) => {
```
Comment (sjaanus) [inline frontend/src/component/onboarding/SdkConnected.tsx:115]: This is the new sdk connected tab, that shows the production ready code. Some parts of code can be removed, when we have 3 layer markdown ready. In next PR.

File: frontend/src/component/onboarding/ConnectSdkDialog.tsx [hunk 4]
```diff
@@ -7,22 +7,23 @@ import {
     useTheme,
 } from '@mui/material';
 import { GenerateApiKey } from './GenerateApiKey';
-import { lazy, Suspense, useEffect, useState } from 'react';
+import { lazy, useEffect, useState } from 'react';
 import { SelectSdk } from './SelectSdk';
 import { GenerateApiKeyConcepts, SelectSdkConcepts } from './UnleashConcepts';
 
 const TestSdkConnection = lazy(() => import('./TestSdkConnection'));
```
Comment (Tymek) [inline frontend/src/component/onboarding/ConnectSdkDialog.tsx:15]:
```suggestion
```

File: frontend/src/component/onboarding/ConnectSdkDialog.tsx [hunk 5]
```diff
@@ -7,22 +7,23 @@ import {
     useTheme,
 } from '@mui/material';
 import { GenerateApiKey } from './GenerateApiKey';
-import { lazy, Suspense, useEffect, useState } from 'react';
+import { lazy, useEffect, useState } from 'react';
```
Comment (Tymek) [inline frontend/src/component/onboarding/ConnectSdkDialog.tsx:10]:
```suggestion
import { useEffect, useState } from 'react';
```

File: frontend/src/component/onboarding/ConnectSdkDialog.tsx [hunk 6]
```diff
@@ -77,13 +78,18 @@ export const ConnectSdkDialog = ({
     const [environment, setEnvironment] = useState<string | null>(null);
     const [apiKey, setApiKey] = useState<string | null>(null);
     const [stage, setStage] = useState<OnboardingStage>('select-sdk');
+    const { project } = useProjectOverview(projectId, {
```
Comment (Tymek) [inline frontend/src/component/onboarding/ConnectSdkDialog.tsx:81]: Isn't that a lot of data? Maybe we should have another endpoint for getting onboarding status?

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 7]
```diff
@@ -70,6 +70,8 @@ export const ProjectFeatureToggles = ({
     const { project } = useProjectOverview(projectId);
     const [connectSdkOpen, setConnectSdkOpen] = useState(false);
 
+    console.log('connectSdkOpen', connectSdkOpen);
+
```
Comment (Tymek) [inline frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureToggles.tsx:74]:
```suggestion
```

**Review by Tymek — APPROVED** [review]

File: frontend/src/component/onboarding/ConnectSdkDialog.tsx (hunk 6 above)
Comment (sjaanus) [reply frontend/src/component/onboarding/ConnectSdkDialog.tsx:79]: It is middle amount. I think it is fine for now, this is still beta. The thing is that this view is opened so rarely, so it makes barely no difference.

**Pull request merged**